tools/roslaunch/src/roslaunch/substitution_args.py

```diff
@@ -346,6 +346,10 @@ def _eval_dirname_context(): return _eval_dirname(context['filename'])
         raise SubstitutionException("$(eval ...) may not contain double underscore expressions")
     return str(eval(s, {}, _DictWrapper(context['arg'], functions)))
 
+def _resolve_eval(resolved, a, unused, context):
+    eval_resolved = _eval(a[5:], context)
+    return resolved.replace("$(%s)" % a, eval_resolved)
```

Comment:
This is how some of the other handlers do this last step as well:

https://github.com/locusrobotics/ros_comm/pull/16/files#diff-85d266eabfa0eb4dda9a4551c1929065R296